models/resourceoperation.go

```diff
@@ -48,19 +53,44 @@ func (ro ResourceOperation) MarshalJSON() ([]byte, error) {
 	if ro.Operation != "" {
 		test.Operation = &ro.Operation
 	}
-	if ro.Object != "" {
+	if ro.DeviceResource != "" {
+		test.DeviceResource = &ro.DeviceResource
+		test.Object = &ro.DeviceResource
+	}
+	if ro.Object != "" && ro.DeviceResource == "" {
 		test.Object = &ro.Object
+		test.DeviceResource = &ro.Object
```

Comment:
The `DeviceResource` check is mutually exclusive here, so this could be written as
```
if ro.DeviceResource != "" {
	test.DeviceResource = &ro.DeviceResource
	test.Object = &ro.DeviceResource
} else if ro.Object != "" {
        test.Object = &ro.Object
        test.DeviceResource = &ro.Object
}
```
Same for `DeviceCommand` below.

Reply:
you are right
modified